code/game/gamemodes/vampire/vampire.dm

```diff
@@ -0,0 +1,486 @@
+	set name = "Recall Thirst"
+	set desc = "Recall how to drain blood from humans. Use if you suddenly cannot do so."
+	set category = "Vampirism"
+
+	if(!is_vampire(usr))
+		usr << "<span class='warning'>The knowledge slips away as you try to grasp it...</span>"
+		usr.verbs -= src
+		return 0
+
+	usr << "<span class='danger'>You recall how to drain blood from humans. If you forget how to do so, use this ability again.</span>"
+	var/datum/martial_art/vampirism/VM = new (null)
+	VM.teach(usr)
+	return 1
+
+//Sanguine Regeneration: Triples healing over time but increases passive blood use.
+/mob/living/carbon/human/proc/vampire_sanguine_regeneration()
+	set name = "Sanguine Regeneration (Toggle)"
+	set desc = "Increases your passive healing ability. Costs 2 cl of clean blood per second or 6 dirty blood per second."
+	set category = "Vampirism"
+
+	if(!is_vampire(usr))
+		usr << "<span class='warning'>The knowledge slips away as you try to grasp it...</span>"
+		usr.verbs -= src
+		return 0
+
+	var/datum/vampire/V = usr.get_vampire()
+	var/mob/living/carbon/human/user = usr
+	V.fast_heal = !V.fast_heal
+	user << "[V.fast_heal ? "<span class='danger'>You begin harnessing your blood to heal your wounds.</span>" : "<span class='danger'>You relax your body's frantic regeneration.</span>"]"
+	return 1
+
+//Accelerated Recovery: Quickly reduces stuns but increases passive blood use.
+/mob/living/carbon/human/proc/vampire_accelerated_recovery()
+	set name = "Accelerated Recovery (Toggle)"
+	set desc = "Dramatically increases stun recovery rate. Costs 3 cl of clean blood per second or 9 dirty blood per second."
+	set category = "Vampirism"
+
+	if(!is_vampire(usr))
+		usr << "<span class='warning'>The knowledge slips away as you try to grasp it...</span>"
+		usr.verbs -= src
+		return 0
+
+	var/datum/vampire/V = usr.get_vampire()
+	var/mob/living/carbon/human/user = usr
+	V.stun_reduction = !V.stun_reduction
+	user << "[V.stun_reduction ? "<span class='danger'>You begin harnessing your blood to empower your metabolism.</span>" : "<span class='danger'>You relax your body's coursing adrenaline.</span>"]"
+	return 1
+
+//Chiropteran Shapeshift: Transforms the vampire into a quick, ventcrawling bat. Return to a normal human at will. Death as a bat or manual cancel revert them to their normal form and stun them.
+/mob/living/carbon/human/proc/vampire_chiropteran_shapeshift()
+	set name = "Chiropteran Shapeshift (100cl)"
+	set desc = "Allows you to morph into a swift bat, capable of using ventilation shafts for movement and passing through creatures. You may revert at will. Costs 100 clean blood or 300 dirty blood."
+	set category = "Vampirism"
+
+	if(!is_vampire(usr))
+		usr << "<span class='warning'>The knowledge slips away as you try to grasp it...</span>"
+		usr.verbs -= src
+		return 0
+
+	var/datum/vampire/V = usr.get_vampire()
+	var/mob/living/carbon/human/user = usr
+	if(!V.use_blood(100, 0))
```

Comment:
Actually for the abilities that use blood, shouldn't you be doing `if(V.use_blood(clean, 1) || V.use_blood(dirty, 0))`